twisted/conch/ssh/keys.py

```diff
@@ -782,18 +783,23 @@ def public(self):
         return Key(self._keyObject.public_key())
 
 
-    def fingerprint(self):
+    def fingerprint(self, hash_function=sha256):
```

Comment:
So I'd like to tag in @reaperhulk or @alex to weigh in on the security risks more formally than me, but here is my limited understanding of the risk here.

The risk only comes in one case: if the fingerprint is used for visual verification by a *user* that the server is the one it is expecting. In this case, an MD5 collision can in principle be used to spoof a server's identity. This is a good reason to change, but it's not clear that the attack surface is huge, especially given that in my experience almost no users actually do this validation anyway. @adiroiban's suggestion that this method is unused within conch itself strongly suggests that there is minimal risk of attack here.

With that said, moving to SHA256 is a really good idea. However, I don't think it's an idea with security ramifications so profound that we need to skip the standard Twisted deprecation process. The security experts I just tagged may disagree with me though.